tools/ui-components/src/button/button.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useMemo } from 'react';
+import { ButtonProps, ButtonSize, ButtonVariant } from './types';
+
+const defaultClassNames = ['cursor-pointer', 'inline-block', 'border-3'];
+
+const computeClassNames = ({
+  size,
+  variant
+}: {
+  size: ButtonSize;
+  variant: ButtonVariant;
+}) => {
+  const classNames = [...defaultClassNames];
+
+  // TODO: support 'link' variant
+  switch (variant) {
+    case 'danger':
+      classNames.push(
+        'border-default-foreground-danger',
+        'bg-default-background-danger',
+        'text-default-foreground-danger'
+      );
+      break;
+    // default variant is 'primary'
+    default:
+      classNames.push(
+        'border-default-foreground-secondary',
+        'bg-default-background-quaternary',
+        'text-default-foreground-secondary'
+      );
+  }
+
+  switch (size) {
+    case 'large':
+      classNames.push('px-4 py-2.5 text-lg');
+      break;
+    case 'small':
+      classNames.push('px-2.5 py-1 text-sm');
+      break;
+    // default size is 'medium'
+    default:
+      classNames.push('px-3 py-1.5 text-md');
+  }
```

Comment:
`px` is horizontal padding, and `py` is vertical padding.

The values are from [Tailwind padding doc](https://tailwindcss.com/docs/padding):
- `{p}-1` is 4px
- `{p}-1.5` is 6px
- `{p}-2.5` is 10px
- `{p}-3` is 12px
- `{p}-4` is 16px

As for the text classes, refer to the changes I made to the `tailwind.config.js` file for their values.

---

Some values here are slightly different from those of the current button component. I have added the rationale to the issue description, under the "Spacing and typography" section, but I'm just copy-pasting it here:

The current values of font size, line height, and padding from Bootstrap seem a bit strange to me, as there are some odd numbers in there, which I think would add some challenges in scaling/positioning. Now that we switch to Tailwind, my proposal is to adopt its system and slightly adjust the styling of our component.

| Size | Current | New |
| --- | --- | --- |
| Small | `padding:  5px 10px` <br> `font-size: 16px`<br> `line-height: 1.5` | `padding:  4px 10px` <br> `font-size: 16px`<br> `line-height: 1.5` |
| Medium | `padding:  6px 12px` <br> `font-size: 18px`<br> `line-height: 1.42857143` | `padding:  6px 12px` <br> `font-size: 18px`<br> `line-height: 1.42857143` |
| Large | `padding:  10px 16px` <br> `font-size: 23px`<br> `line-height: 1.3333333` | `padding:  10px 16px` <br> `font-size: 24px`<br> `line-height: 1.3333333` |